### ISSUE TITLE
Regression in .NET 9 RC2: IDE0052 reports false positive with source-generated method 

### ISSUE THREAD

**Issue by bart-vmware** [dotnet/roslyn#75483]
**Version Used**: 
.NET 9 RC2

**Steps to Reproduce**:
After updating from .NET 8 to .NET 9 RC2, we're getting a false positive on [this line](https://github.com/SteeltoeOSS/Steeltoe/blob/try-net90/src/Configuration/src/Encryption/DecryptionConfigurationProvider.cs#L16-L18):

```c#
internal sealed partial class DecryptionConfigurationProvider(
    IList<IConfigurationProvider> providers, ITextDecryptor? textDecryptor, ILoggerFactory loggerFactory)
    : CompositeConfigurationProvider(providers, loggerFactory)
{
#pragma warning disable IDE0052 // Remove unread private members
    private readonly ILogger<DecryptionConfigurationProvider> _logger = loggerFactory.CreateLogger<DecryptionConfigurationProvider>();
#pragma warning restore IDE0052 // Remove unread private members

    // ...

    [LoggerMessage(Level = LogLevel.Trace, Message = "Decrypted value '{CipherValue}' at key '{Key}' to '{PlainTextValue}'.")]
    private partial void LogDecrypt(string key, string? cipherValue, string? plainTextValue);
}
```

The private field is used by the source-generated `LogDecrypt` logger method, so the warning should not appear.

To reproduce, clone the repo at https://github.com/SteeltoeOSS/Steeltoe, check out branch `try-net90`, open `src/Steeltoe.All.sln` in Visual Studio 2022 v17.12 (latest preview) and build.

**Diagnostic Id**:
IDE0052

**Expected Behavior**:
No warning, which is the behavior in .NET 8.

**Actual Behavior**:
A warning appears during build.

**Comment by CyrusNajmabadi** [comment]
Closing as behavior in the presense of SG docs is 'best effort' and we opt for a lower fidelity understanding mode to ensure acceptable performance.

**Comment by bart-vmware** [comment]
@CyrusNajmabadi Can you please elaborate on what's going on here? I'm shocked to see this closed as not planned.

We recently updated a small part of our codebase to use the logger source generator, to discover how that works out. IDE0052 worked fine on .NET 8, but breaks on .NET 9 as reported. Does this mean we shouldn't be using the logger source generator, or even any source generators at all? Suppressing thousands of these false positives in the codebase is a no-go for us, so we'll likely just turn off IDE0052 entirely. 

Are there more roslyn-provided analyzers that break in the presence of source generators? Is it documented anywhere that roslyn analyzers don't work reliably when source generators are used, even with the ones provided by the .NET runtime itself?

**Comment by CyrusNajmabadi** [comment]
At this point, except for security analyzers, we basically don't run analyzers on generators by default.  You can see that default here:

https://github.com/dotnet/roslyn/blob/6f2ed01d1af9429d874aca1eea60d3212167bc40/src/Analyzers/Core/Analyzers/AbstractBuiltInCodeStyleDiagnosticAnalyzer_Core.cs#L69-L73

> so we'll likely just turn off IDE0052 entirely.

That's definitely very reasonable.  If an analyzer isn't providing you valuable, def disable.

> Does this mean we shouldn't be using the logger source generator, or even any source generators at all? 

You should def use SGs if they're providing you with value.  But the IDE experience is something that is greatly in flux, with lots of changes driven by the needs of the community (esp. in the area of perf).

For context, for perf and stability we run generators in a different process.  These generators now run in isolated assembly load contexts so that they also can be reloaded when they change on disk.  As part of the perf work though, we had to do a lot of adjustments.  Including moving features so that they often will run even with SG data is not available.  This was because the prior system would wait until SG data was ready which caused delays across the board for virtually all users using SGs.

Note: if you want to use the previous experience, you can go back to it with:

![Image](https://github.com/user-attachments/assets/667b79f6-2931-4fec-9b48-429f256d54df)

Note: the perf impact of this can be significant, depending on the generators you have and the particulars of your project.

**Comment by bart-vmware** [comment]
@CyrusNajmabadi Thanks for providing context, makes more sense now.

We use the following MSBuild properties to fail command-line builds on code style violations during PR builds:

```xml
<PropertyGroup>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
    <AnalysisMode>Recommended</AnalysisMode>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
</PropertyGroup>
```

with this in `.editorconfig`:

```ini
dotnet_analyzer_diagnostic.category-Style.severity = warning
```

Aside from changing Visual Studio options, is there a way (MSBuild property, environment variable, etc) to revert to the old behavior in command-line builds? We value correctness and predictability over performance, especially there.
